### PR TITLE
[mbedtls] Update to 3.6.1

### LINKED ISSUE #25703: [mbedtls] update to 3.2.1
Library name: mbedtls

New version number: 3.2.1

Other information that may be useful (release notes, etc...)
https://github.com/Mbed-TLS/mbedtls/releases
https://github.com/Mbed-TLS/mbedtls-docs/blob/main/security-advisories/advisories/mbedtls-security-advisory-2022-07.md

### PR DESCRIPTION
**Pull request by dg0yt**
Resolves #25703.
Cleanup, simplify.
Enable uwp.

### CODE REVIEW

**Comment by talregev** [comment]
I can close my PR.

**Comment by dg0yt** [comment]
Depends on #40677.

**Comment by talregev** [comment]
version 3.6.1 is out.

File: ports/mbedtls/usage
Comment (WangWeiLin-MV) [inline ports/mbedtls/usage:1]: The port usage tests pass with `x64-linux`.

**Review by WangWeiLin-MV — APPROVED** [review]
The ports installation tests pass with the following triplets:

* x64-linux

**Pull request merged**